src/components/login/login.spec.js

```diff
@@ -0,0 +1,145 @@
+
+const authReducer = {
+  isAuthenticated: false,
+  user: {},
+  status: 'rest'
+};
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+const store = mockStore({
+  authReducer
+});
+
+const props = {
+  LoginForm: jest.fn()
+};
+
+const unconnectedProps = {
+  authAction: jest.fn(),
+  status: '',
+  isAuthenticated: false,
+  error: null,
+  location: { url: 'http://localhost' }
+};
+
+const setUp = () =>
+  mount(
+    <Provider store={store} {...props}>
+      <Router>
+        <LoginForm />
+      </Router>
+    </Provider>
+  );
+
+describe('Login Component', () => {
+  it('should render without crashing', () => {
+    const wrapper = setUp();
+    expect(wrapper).toBeDefined();
+    expect(wrapper.find('LoginForm')).toBeTruthy();
+    expect(wrapper.find('div')).toBeTruthy();
+  });
+
+  it('should render one form', () => {
+    const wrapper = setUp();
+    expect(wrapper.find('form').length).toBe(1);
+  });
+
+  it('should render 3 input fields', () => {
+    const wrapper = setUp();
+    expect(wrapper.find('input').length).toBe(3);
+    expect(
+      wrapper
+        .find('input')
+        .at(0)
+        .props().type
+    ).toBe('email');
+    expect(
+      wrapper
+        .find('input')
+        .at(1)
+        .props().type
+    ).toBe('password');
+    expect(
+      wrapper
+        .find('input')
+        .at(2)
+        .props().type
+    ).toBe('submit');
+  });
+
+  it('should redirect to Profile if user is logged in already', () => {
+    authReducer.isAuthenticated = true;
+    const wrapper = setUp();
+    expect(wrapper).toBeDefined();
+    authReducer.isAuthenticated = true;
+    expect(wrapper.find('Redirect').length).toBe(1);
+    expect(wrapper.find('Redirect').props().to).toBe('/profile');
+    expect(wrapper.find('div').length).toBe(0);
+  });
+
+  it('should not render a loading component when status is at rest', () => {
+    authReducer.status = 'rest';
+    authReducer.isAuthenticated = false;
+    const wrapper = setUp();
+    expect(wrapper).toBeDefined();
+    expect(wrapper.find('Loader')).toBeTruthy();
+    expect(wrapper.find('Loader').props().loading).toBe(false);
+  });
+
+  it('should render a loading component when status equals authenticationPending', () => {
+    authReducer.status = 'authenticationPending';
+    authReducer.isAuthenticated = false;
+    const wrapper = setUp();
+    expect(wrapper).toBeDefined();
+    expect(wrapper.find('Loader')).toBeTruthy();
+    expect(wrapper.find('Loader').props().loading).toBe(true);
+  });
+
+  it('should not submit form when email or pass is empty', () => {
+    const wrapper = shallow(<UnconnectedLoginForm {...unconnectedProps} />);
+    const event = {
+      preventDefault: jest.fn()
+    };
+
+    wrapper.find('form').simulate('submit', event);
+    expect(wrapper.state('errors').email).toEqual('Email cannot be empty');
+    expect(wrapper.state('errors').password).toEqual(
+      'Password cannot be empty'
```

Comment:
Test that the errors actually are displayed in the DOM not just in state.

Checking the value in state is testing implementation detail. Don't do that!